scripts/trace_docs_deps.py

```diff
@@ -0,0 +1,110 @@
+"""
+Usage: `python trace_docs_deps.py [DOCS_DIR]`
+
+This script is used in our Azure Docs Integration pipeline (azure-pipelines-docs-integration.yml) to determine whether
+a change has been made in the `great_expectations/` directory that change impacts `docs/` and the snippets therein.
+
+The script takes the following steps:
+    1. Parses all markdown files in `docs/`, using regex to find any Docusaurus links (i.e. ```python file=...#L10-20)
+    2. Goes to each linked file and uses AST to parse imports used there
+    3. Filters for only relative imports and determines the paths to those files
+
+The resulting output list is all of the dependencies `docs/` has on the primary `great_expectations/` directory.
+If a change is identified in any of these files during the pipeline runtime, we know that a docs dependency has possibly
+been impacted and the pipeline should run to ensure adequate test coverage.
+
+Please note that this solution has some inherit disclaimers one should be familiar with. As there is no actual running
+of source code (but rather parsing of the source file's AST), it is hard to exactly pinpoint the location of a given
+import. Utilizing the `__file__` attr would provide the exact path of an import but without that, we need to assume that
+an ast.Import node corresponds to an accurate path in our codebase (i.e. `great_expectations.core.batch` results in
+great_expectations/core/batch.py).
+
+As long as imports correspond to the relative path of modules, this method will continue to work as intended. Deviations
+here would result in certain imports not being picked up by the script, reducing the accuracy and frequency of the
+Azure Docs Integration pipeline's triggers. As GE has maintained this consistency throughout its development, this
+script and the methods used herein are deemed to be appropriate.
+
+"""
+
+import ast
+import glob
+import os
+import re
+import sys
+from typing import List, Set
+
+from great_expectations.data_context.util import file_relative_path
+
+
+def find_docusaurus_refs(dir: str) -> List[str]:
+    """ Finds any Docusaurus links within a target directory (i.e. ```python file=...#L10-20) """
+    linked_files: Set[str] = set()
+    pattern: str = (
+        r"\`\`\`[a-zA-Z]+ file"  # Format of internal links used by Docusaurus
+    )
+
+    for doc in glob.glob(f"{dir}/**/*.md", recursive=True):
+        for line in open(doc):
+            if re.search(pattern, line):
+                file: str = _parse_file_from_docusaurus_link(line)
+                path: str = file_relative_path(doc, file)
```

Comment:
I think this should be `os.path.join(os.path.dirname(doc), file)` (what `file_relative_path` is actually doing) so we can remove GE as a dependency from this script.